Anchor/alias handling broken (related to LoadBlessed?)

Anchors/aliases seem to be broken now. Sample code:
```
#!/usr/bin/perl

use strict;
use warnings;
use Data::Dumper;

my $recursive_schema = <<'EOF';
mapping:   &main-rule
  mapping:   *main-rule
EOF

for my $mod (qw(YAML YAML::PP YAML::XS YAML::Syck)) {
    eval 'require '.$mod; die $@ if $@;
    warn "$mod " . eval('$'.$mod.'::VERSION') . "...\n";
    my $func = do {
	no strict 'refs';
	\&{$mod.'::Load'};
    };
    #eval('$'.$mod.'::LoadBlessed = 1'); die $@ if $@;

    warn Dumper($func->($recursive_schema));
}
__END__
```

Output:
```
YAML 1.30...
$VAR1 = {
          'mapping' => {
                         'mapping' => $VAR1->{'mapping'}
                       }
        };
YAML::PP 0.018...
$VAR1 = {
          'mapping' => {
                         'mapping' => $VAR1->{'mapping'}
                       }
        };
YAML::XS 0.81...
$VAR1 = {
          'mapping' => {
                         'mapping' => $VAR1->{'mapping'}
                       }
        };
YAML::Syck 1.32...
$VAR1 = {
          'mapping' => {
                         'mapping' => {
                                        'name' => 'main-rule'
                                      }
                       }
        };
```
Workaround is to set `$YAML::Syck::LoadBlessed=1` --- however, there are no blessed references involved here.

Trying that approach breaks tests (possibly correctly?)
```
Test Summary Report
-------------------
t/2-scalars.t             (Wstat: 512 Tests: 134 Failed: 2)
  Failed tests:  48-49
  Non-zero exit status: 2
t/yaml-alias.t            (Wstat: 256 Tests: 14 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
Files=28, Tests=619,  4 wallclock secs ( 0.20 usr  0.06 sys +  3.18 cusr  0.53 csys =  3.97 CPU)
Result: FAIL
Failed 2/28 test programs. 3/619 subtests failed.
make: *** [test_dynamic] Error 255
```

--- a/perl_syck.h
+++ b/perl_syck.h
@@ -554,8 +554,7 @@ yaml_syck_parser_handler
                 if (hv_store_ent(map, key, val, 0) != NULL)
                     USE_OBJECT(val);
                 sv = newRV_noinc((SV*)map);
-                if ( load_blessed )
-                    sv_bless(sv, gv_stashpv("YAML::Syck::BadAlias", TRUE));
+                sv_bless(sv, gv_stashpv("YAML::Syck::BadAlias", TRUE));
             }
             else
 #endif

--- a/t/2-scalars.t
+++ b/t/2-scalars.t
@@ -1,7 +1,7 @@
 use FindBin;
 BEGIN { push @INC, $FindBin::Bin }
 
-use TestYAML tests => 134;
+use TestYAML tests => 135;
 
 ok( YAML::Syck->VERSION, "YAML::Syck has a version and is loaded" );
 
@@ -184,6 +184,18 @@ Zort:\s&(?!\1)(\d+)\s*\n
 }xms;
 
 like( Dump( scalar Load($recurse2) ), $recurse2want, 'recurse 2' );
+
+my $recurse3 = << '.';
+cycle: &cycle
+  a: *cycle
+.
+
+{
+    local $YAML::Syck::LoadBlessed = 0;
+    my $data = Load($recurse3);
+    is("$data->{cycle}", "$data->{cycle}->{a}", 'Circular refs are working correctly');
+}
+
 
 is( Dump( 1, 2, 3 ), "--- 1\n--- 2\n--- 3\n" );
 is( "@{[Load(Dump(1, 2, 3))]}", "1 2 3" );

